js/gated-content/src/components/MediaPlayer.vue

```diff
@@ -113,6 +125,27 @@ export default {
       this.playbackInProgress = false;
       this.handlePlayerEvent('videoPlaybackEnded');
     },
+    setTimeCode(player) {
+      if (this.player == 'vimeo') {
+        const timecode = this.media.field_media_video_embed_field.substring(
+          this.media.field_media_video_embed_field.lastIndexOf('#t=') + 3,
+          this.media.field_media_video_embed_field.lastIndexOf('s')
+        );
+        if (!isNaN(timecode)) {
+          const timecodeInSeconds = parseInt(timecode, 10);
+          this.$refs.player.player.setCurrentTime(timecodeInSeconds);
+        }
+      } else {
+        const timecode = this.media.field_media_video_embed_field.substring(
+          this.media.field_media_video_embed_field.lastIndexOf('#t=') + 3,
+          this.media.field_media_video_embed_field.lastIndexOf('s')
+        );
+        if (!isNaN(timecode)) {
+          const timecodeInSeconds = parseInt(timecode, 10);
```

Comment:
Any reason this is done inside each section of the `if`?
Probably we should do just the last lines (player seek to the needed timecode) depending on which player is rendered?